src/esn.calendar.libs/app/components/event-date-suggestion/event-date-suggestion.controller.js

```diff
@@ -79,4 +83,41 @@ function calEventDateSuggestionController(esnI18nDateFormatService, calMoment, e
     }
     self.diff = self.event.end.diff(self.event.start);
   }
+
+  // Only fired when using the native mobile picker.
+  function onStartDateTimeChange() {
+    self.event.start.set({
+      hour: self.startTime.getHours(),
+      minute: self.startTime.getMinutes()
+    });
+
+    self.onStartDateChange();
+    // Update the input fields to display the new time ( in case of any internal change like offset ).
+    _initTimeInputs();
+  }
+
+  // Only fired when using the native mobile picker.
+  function onEndDateTimeChange() {
+    self.event.end.set({
+      hour: self.endTime.getHours(),
+      minute: self.endTime.getMinutes()
+    });
+
+    self.onEndDateChange();
+    // Update the input fields to display the new time ( in case of any internal change like offset ).
+    _initTimeInputs();
+  }
+
+  function _initTimeInputs() {
+    if (!self.event.start || !self.event.start.isValid() || !self.event.end || !self.event.end.isValid()) {
+      return;
+    }
+
+    self.startTime = self.event.start.toDate();
+    self.endTime = self.event.end.toDate();
+
+    // Set the hours to avoid the timzone issues when converting a moment object to Date.
+    self.startTime.setHours(self.event.start.hours());
+    self.endTime.setHours(self.event.end.hours());
```

Comment:
1. First, I think this logic should only impact mobile devices. From what I read here, it also affects every other device.
2. If I recall correctly, we have some timezones that are only +/- 30 minutes away from each other, so only setting hours may not be enough.

Reply:
I agree with you on `2`, it's fixed now, we set the minutes also.
for `1`, `_setTimeInputs` is only related to the input models `startTime` and `endTime` that are only used with the native `time` fields ( that only shows for mobile ).

Reply:
I see, how about `_initMobileTimeInputs` for the function's name?

Reply:
done